pkg/pgxconn/pgx_conn.go

```diff
@@ -109,15 +147,53 @@ func (p *connImpl) Close() {
 }
 
 func (p *connImpl) Exec(ctx context.Context, sql string, args ...interface{}) (pgconn.CommandTag, error) {
-	return p.Conn.Exec(ctx, sql, args...)
+	lset := getLabelSet("exec")
+	requestTotal.With(lset).Inc()
+	start := time.Now()
+	defer func() {
+		duration.With(lset).Observe(time.Since(start).Seconds())
```

Comment:
How about adding sql query as a label? I know that `sql` will increase series cardinality but it will give much better insight. `duration` that we measure is generic and mostly depends on query that is running. The same question goes for other conn functions.

Reply:
We may add this as part of trace implementation in pgx/database module. This could be as event or log. cc @antekresic